socket.c

```diff
@@ -1055,35 +1025,42 @@ int nasp_connect(struct sock *sk, struct sockaddr *uaddr, int addr_len)
 		prot = &nasp_v6_prot;
 	}
 
+	if (err != 0)
+	{
+		goto error;
+	}
+
 	printk("nasp: nasp_connect uid: %d app: %s to port: %d", current_uid().val, current->comm, port);
 
-	opa_socket_context opa_socket = socket_eval(port, OUTPUT, current->comm, current_uid().val);
-	if (err == 0 && opa_socket.allowed)
+	nasp_socket *sc = nasp_socket_connect(sk);
+	if (!sc)
 	{
-		const char *server_name = NULL; // TODO, this needs to be sourced down here
+		pr_err("nasp: nasp_socket_connect failed to create nasp_socket");
+		goto error;
+	}
 
-		nasp_socket *sc = nasp_socket_connect(sk);
-		if (!sc)
-		{
-			pr_err("nasp: nasp_socket_connect failed to create nasp_socket");
-			goto error;
-		}
+	// send attest command
+	command_answer *answer = send_attest_command(OUTPUT, sk, port);
 
-		memcpy(sc->rsa_priv, rsa_priv, sizeof *sc->rsa_priv);
-		memcpy(sc->rsa_pub, rsa_pub, sizeof *sc->rsa_pub);
+	if (answer->error)
+	{
+		pr_err("nasp: connect failed to send attest command: %s", answer->error);
```

Comment:
Same as above, shouldn't we goto error in this case?

Reply:
This is a design decision made by us to fall through without mtls when the attestation fails for some reason.